pandas/core/internals/blocks.py

```diff
@@ -734,14 +734,25 @@ def replace(
 
             # try again with a compatible block
             block = self.astype(object)
-            return block.replace(
+            block_replaced = block.replace(
                 to_replace=to_replace,
                 value=value,
                 inplace=inplace,
                 filter=filter,
                 regex=regex,
                 convert=convert,
             )
+            blocks_converted = []
+            for ls_elem in block_replaced:
```

Comment:
why is this logic not just in Block.replace itself? seems too complicated here

Reply:
I wanted to ensure, that the conversion would only take place, after the block was forcefully cast to an object before.

I can add it to the replace function, if that would be better.

Reply:
> I wanted to ensure, that the conversion would only take place, after the block was forcefully cast to an ObjectBlock before.
> 
> I can add it to the replace function, if that would be better.

the regression was caused by a clean-up to Block.replace (#27768), see https://github.com/pandas-dev/pandas/issues/32988#issuecomment-605519281

so fixing Block.replace as @jreback suggests seems more appropriate